### PR TITLE
BL-1071: Disable jquery.longpress if an input processor is being used on Windows

### PR DESCRIPTION
**Pull request by phillip-hopper**

### CODE REVIEW

File: src/BloomBrowserUI/bookEdit/js/bloomEditing.js [hunk 1]
```diff
@@ -1541,9 +1541,12 @@ function SetupElements(container) {
         }
     });
 
-    $(container).find('.bloom-editable').longPress();
+    getIframeChannel().simpleAjaxGet('/bloom/windows/useLongpress', function(response) {
+        if (response === 'Yes')
+            $(container).find('.bloom-editable').longPress();
+    });
```
Comment (phillip-hopper) [inline src/BloomBrowserUI/bookEdit/js/bloomEditing.js]: This is the only non-white-space change in this file.

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/287/
Test FAILed.

**Comment by hatton** [comment]
Shall we we put the TFS part in libpalaso?

File: src/BloomExe/MsTsfInterfaces.cs [hunk 2]
```diff
@@ -0,0 +1,352 @@
+﻿// Copyright (c) 2013 SIL International
+// This software is licensed under the MIT License (http://opensource.org/licenses/MIT)
+#if !__MonoCS__
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Unmanaged.TSF
```
Comment (JohnThomson) [inline src/BloomExe/MsTsfInterfaces.cs]: Are we supposed to use the Microsoft namespace? Even for redefining their interfaces?

File: src/BloomExe/MsTsfInterfaces.cs [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+{
+	#region Structs
+	[StructLayout(LayoutKind.Sequential, Pack = 4)]
+	internal struct TfInputProcessorProfile
```
Comment (JohnThomson) [inline src/BloomExe/MsTsfInterfaces.cs]: Is there a source we should acknowledge for all this tricky detail?

**Comment by phillip-hopper** [comment]
@hatton It is in Palaso, that's where I found it. The problem is that it is "internal."

File: src/BloomExe/MsTsfInterfaces.cs (hunk 2 above)
Comment (phillip-hopper) [inline src/BloomExe/MsTsfInterfaces.cs]: I just copied the entire file from Palaso, so I can't say if it should be that way.

File: src/BloomExe/web/EnhancedImageServer.cs [hunk 4]
```diff
@@ -192,6 +200,27 @@ protected override bool ProcessRequest(IRequestInfo info)
 				if (ProcessI18N(localPath, info))
 					return true;
 			}
+			else if (localPath.StartsWith("windows/useLongpress"))
+			{
+				// On Linux we need to just let this stop here.
+				if (!Palaso.PlatformUtilities.Platform.IsWindows) return true;
```
Comment (JohnThomson) [inline src/BloomExe/web/EnhancedImageServer.cs]: Shouldn't we construct the "Yes"/"No" response no matter what?

File: src/BloomExe/web/EnhancedImageServer.cs [hunk 5]
```diff
@@ -192,6 +200,27 @@ protected override bool ProcessRequest(IRequestInfo info)
...
+
+				// In order to detect an input processor, we need to execute this on the main UI thread.
+				var frm = Application.OpenForms.Cast<Form>().FirstOrDefault(f => f is Shell);
+				if (frm == null) return false;
```
Comment (JohnThomson) [inline src/BloomExe/web/EnhancedImageServer.cs]: Construct response "no"?

File: src/BloomExe/web/EnhancedImageServer.cs (hunk 4 above)
Comment (phillip-hopper) [inline src/BloomExe/web/EnhancedImageServer.cs]: Yes, my bad.

**Comment by hatton** [comment]
@phillip-hopper well, good thing we own libpalaso, so we can change it to "Public". But more likely, you just need to find a proper home for a single public static method  to answer the question you're after.

**Comment by JohnThomson** [comment]
Seems that your IsUsingInputProcessor method could be public static in PalasoUiWindowsForms somewhere, probably with a Form argument and adding some of the logic to (a) answer false if the form is null, (b) invoke if we're in the wrong thread for that form. Maybe just on KeyboardController?

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/292/
Test FAILed.

**Comment by phillip-hopper** [comment]
Pull request updated, but now is waiting on Palaso PR #197 to be merged first.

**Comment by phillip-hopper** [comment]
retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/293/
Test FAILed.

**Comment by phillip-hopper** [comment]
retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/294/
Test FAILed.

**Comment by phillip-hopper** [comment]
retest this please

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/295/
Test FAILed.

**Comment by sillsdevgerrit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.lsdev.sil.org//job/GitHub-Bloom-Wrapper-master-debug/297/
Test PASSed.

**Pull request merged**